pkg/providers/aws/config.go

```diff
@@ -26,7 +26,7 @@ import (
 const (
 	DefaultConfigMapName = "cloud-resources-aws-strategies"
 
-	DefaultFinalizer = "finalizers.cloud-resources-operator.integreatly.org"
+	DefaultFinalizer = "cloud-resources-operator.integreatly.org/finalizers"
```

Comment:
@makslion we will need to handle this change somehow because when this change lands on an existing installation we won't be able to remove/reconcile finalizers.
So I think everywhere where we reconcile finalizers we need to check finalizer format > update if it's in old format.
Let me know if you have any questions, sample how this was done in intly: https://github.com/integr8ly/integreatly-operator/pull/1807/files

Reply:
Seems like I can't verify it fast enough, so will just put here a theory on this. In the [integreatly operator](https://github.com/integr8ly/integreatly-operator/pull/1807/files#diff-239eca6331989433c0101fe047f78aa8ce2dab5047a733719972e99206a6d9faL315) we are setting finalizers by retrieving existing ones and appending them with a new one and then calling an `.Update` on the object. So if there was an old and not used finalizer it would be left there, hence the need in replacing it.
In the CRO it is a bit funny. If I'm not going crazy, we don't have any sort of `reconcileFinalizer` but instead, we are calling the [CreateFinalizer](https://github.com/integr8ly/cloud-resource-operator/blob/master/pkg/resources/finalizers.go#L48-L63) each loop and the "setting" is done by the [addFinalizer](https://github.com/integr8ly/cloud-resource-operator/blob/master/pkg/resources/finalizers.go#L19-L23) function. The `addFinalizer` checks for the existence of the finalizer we want and then if it is not found, we are reassigning that field. So in case when we will have an old finalizer it would not find the new one and will call the [setFinalizers](https://github.com/makslion/cloud-resource-operator/blob/MGDAPI-1635/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/meta.go#L170) that would reassign (hence, replace) finalizer to be the new one. :thinking: